common/cpp/crypto/verify_signature.cpp

```diff
@@ -47,13 +48,14 @@ bool verify_signature(const char* cert_pem,
     ret = EVP_VerifyUpdate(ctx, msg, msg_len);
     assert(ret == 1);
 
-    int signature_decoded_len = 2048;
-    unsigned char signature_decoded[signature_decoded_len];
-    ret = tcf::crypto::EVP_DecodeBlock_wrapper(signature_decoded,
-                                  signature_decoded_len,
-                                  (unsigned char*)signature,
-                                  signature_len);
-    assert(ret!=-1);
+    // Decode base64-encoded signature.
+    // Decoded output may have 1-3 extra '\0' bytes appended.
+    static const int signature_decoded_len = 2048;
+    unsigned char signature_decoded[signature_decoded_len + 3];
+    ret = EVP_DecodeBlock(signature_decoded,
+                          (unsigned char*)signature,
+                          signature_len + 3);
```

Comment:
Extra buffer size to allocate \o characters was not there when we are using the EVP_DecodeBlock_wrapper() function. Was that a miss?

Reply:
No. That was avoided by having EVP_DecodeBlock_wrapper(), which I removed. The only function it did was trim the extra \0 characters.

Reply:
Ok.